src/stan/mcmc/hmc/xhmc/base_xhmc.hpp

```diff
@@ -283,7 +284,7 @@ namespace stan {
       double x_delta_;
 
       int n_leapfrog_;
-      int divergent_;
+      bool divergent_;
```

Comment:
For future reference we recommend PRs isolate one fix at a time and avoid aggregating small, unrelated updates like this.

Reply:
Ok, just fixed problems discussed in related issue page.